### PR TITLE
Fix the consistency of pre-processing with yolov5

### LINKED ISSUE #274: Make the pre-processing consistent with yolov5
### 🚀 The feature and motivation
As mentioned in https://github.com/zhiqwang/yolov5-rt-stack/issues/265#issuecomment-1008351793, the `YOLOv5.load_from_yolov5()` strategy will introduce minor differences with ultralytics, now there are two ways to solve this problem:

1. Make the [pre-processing](https://github.com/zhiqwang/yolov5-rt-stack/blob/949fa0b08dd336c21dc051594941af457a312b21/yolort/models/transform.py#L38) behave the same with yolov5. The advantage of this approach is that there is no need to write code on the C++ backend.
2. Just use `YOLO.load_from_yolov5()` and implement the pre-precessing in the C++ backend. This requires to modify the example we provide. And the main difference between this strategy and the official yolov5 is that we don't need to write the post-processing (`nms`) module on the C++ backend.

In fact, the main ops needed for preprocessing is `interpolate`, yolov5 uses the `cv2.resize` operator, we use `torch.nn.functional.interpolate` in yolort. (And the ops in cv2 can't be traced or scripted.) The implementation of the `interpolate` operator is not entirely consistent across frameworks. We will optimize the implementation of our pre-processing this week and hope that the results will be available next week.

https://github.com/zhiqwang/yolov5-rt-stack/blob/949fa0b08dd336c21dc051594941af457a312b21/yolort/v5/utils/augmentations.py#L126

https://github.com/zhiqwang/yolov5-rt-stack/blob/949fa0b08dd336c21dc051594941af457a312b21/yolort/models/transform.py#L286-L293

### Additional context
https://pytorch.org/blog/introducing-torchvision-new-multi-weight-support-api/#associated-meta-data---preprocessing-transforms

_Originally posted by @zhiqwang in https://github.com/zhiqwang/yolov5-rt-stack/issues/271#issuecomment-1009600778 cc @mattpopovich ._

### PR DESCRIPTION
**Pull request by q3394101**
Aligning the `YOLOTransform` with the `letterbox` in yolov5. Close #274 .

https://github.com/zhiqwang/yolov5-rt-stack/blob/c6131f438f8981fcf43e1dec1a6beda22fb19cc7/yolort/models/transform.py#L38-L52

https://github.com/zhiqwang/yolov5-rt-stack/blob/c6131f438f8981fcf43e1dec1a6beda22fb19cc7/yolort/v5/utils/augmentations.py#L96-L105

### CODE REVIEW

File: test/test_models_transform.py [hunk 1]
```diff
@@ -19,3 +22,32 @@ def test_yolo_transform():
...
+    im3, _ = _resize_image_and_masks(im3.float(), new_shape)
+    out1 = yolo_transform.batch_images([im3])
+
+    out2 = letterbox(img_numpy, new_shape=new_shape, auto=auto, stride=stride)
+
+    aug1 = out1[0].numpy()
+    aug2 = out2[0].astype(np.float32)  # uint8 to float32
+    aug2 = np.transpose(aug2 / 255.0, [2, 0, 1])
+    assert aug1.shape == aug2.shape
+    np.testing.assert_allclose(aug1, aug2, rtol=1e-4, atol=1e-2)
```
Comment (zhiqwang) [inline test/test_models_transform.py:53]: PyTorch's [interpolate](https://pytorch.org/docs/stable/generated/torch.nn.functional.interpolate.html#torch-nn-functional-interpolate) operator now only aligns with OpenCV for the float type, and the `letterbox` implemented by yolov5 operates on integers, so we set a lower precision here. Check https://github.com/pytorch/pytorch/issues/5580#issuecomment-907051670 for more details.

**Review by zhiqwang — APPROVED** [review]
